### PR TITLE
doc: describe CephFS max_file_size

### PR DESCRIPTION
**Pull request by ktdreyer**
Add a description of max_file_size to the CephFS admin docs.

Thanks to John Spray <jspray@redhat.com> on ceph-users for this information.

### CODE REVIEW

File: doc/cephfs/administration.rst [hunk 1]
```diff
@@ -42,6 +42,34 @@ creation of multiple filesystems use ``ceph fs flag set enable_multiple true``.
...
+CephFS has a configurable maximum file size, and it's 1TB by default.
+You may wish to set this limit higher if you expect to store large files
+in CephFS. It is a 64-bit field.
+
+CephFS enforces this limit at the point of appending to files or
+setting their size. It does not affect how anything is stored.
+
+When users create a file of an enormous size (without necessarily
+writing any data to it), some operations (such as deletes) cause the MDS
+to have to do a ridiculously large number of operations to check if any
```
Comment (alfredodeza) [inline doc/cephfs/administration.rst]: I don't think you need to point out that the number is ridiculously large. s/ridiculously//g reads OK to me

**Comment by jcsp** [comment]
This neatly captures what I wrote on the mailing list, but it's perhaps too much detail for the user docs -- the middle three paragraphs are more of an architectural explanation of why the field exists, it could confuse people.

If you want to keep that text, I'd bump it to the bottom or put it in an info/background sidebar.

**Comment by vasukulkarni** [comment]
:+1:

**Comment by ktdreyer** [comment]
@jcsp I've moved the paragraphs to a new "Maximum file sizes and performance" section. Trying to help users understand the implications of raising the limit.

**Comment by jcsp** [comment]
:+1: just needs a Signed-off-by line on the commit that doesn't have one.

**Pull request merged**